hasjob/views/listing.py

```diff
@@ -972,3 +957,62 @@ def newjob():
     # 2. POST request from main page's Post a Job box
     # 3. POST request from this page, with errors
     return render_template('postjob.html', form=form, no_removelogo=True)
+
+
+@app.route('/<domain>/<hashid>/close', methods=('GET', 'POST'), defaults={'key': None}, subdomain='<subdomain>')
+@app.route('/<domain>/<hashid>/close', methods=('GET', 'POST'), defaults={'key': None})
+@app.route('/close/<hashid>', methods=('GET', 'POST'), defaults={'key': None, 'domain': None}, subdomain='<subdomain>')
+@app.route('/close/<hashid>/<key>', defaults={'domain': None}, methods=('GET', 'POST'), subdomain='<subdomain>')
+@app.route('/close/<hashid>', methods=('GET', 'POST'), defaults={'key': None, 'domain': None})
+@app.route('/close/<hashid>/<key>', defaults={'domain': None}, methods=('GET', 'POST'))
```

Comment:
Don't use `/close/<hashid>/*` since it's not a legacy URL. Use `/<domain>/<hashid>/close/*`.